File: mcci/src/main/java/com/evernightfireworks/mcci/CausalEngine.java
package com.evernightfireworks.mcci;

import com.evernightfireworks.mcci.blocks.CausalBlocks;
import com.evernightfireworks.mcci.services.CraftingPolicyService;
import net.fabricmc.api.ModInitializer;


import net.minecraft.util.Identifier;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class CausalEngine implements ModInitializer {
    private static final Logger logger = LogManager.getFormatterLogger(CausalEngine.class.getName());
    public static CraftingPolicyService CRAFTING_POLICY_SERVICE;

    @Override
    public void onInitialize() {
        logger.info("Causal engine initializing...");
        CausalBlocks.registerMain();
        CraftingPolicyService.registerMain();
        CRAFTING_POLICY_SERVICE = new CraftingPolicyService();
        logger.info("Causal engine initialized");
    }
}

File: mcci/src/main/kotlin/com/evernightfireworks/mcci/CausalEngine.kt
package com.evernightfireworks.mcci

import net.fabricmc.api.ModInitializer
import net.minecraft.item.Item
import net.minecraft.item.ItemGroup
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry
import org.apache.logging.log4j.LogManager

class CausalEngine:  ModInitializer {
    companion object {
        val CAUSAL_MACHINE = CausalMachine(Item.Settings().group(ItemGroup.MISC).maxCount(1))
    }
    val logger = LogManager.getFormatterLogger("KotlinLanguageTest")
    override fun onInitialize() {
        logger.info("Hello from causal engine")
        Registry.register(Registry.ITEM,
                Identifier("mcci", "causal_machine"), CAUSAL_MACHINE);
    }
}